ChemistryLib/PhreeqcInterfaceData/EquilibriumPhase.cpp

```diff
@@ -0,0 +1,63 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2019, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "EquilibriumPhase.h"
+#include "BaseLib/ConfigTreeUtil.h"
+
+namespace ChemistryLib
+{
+std::vector<EquilibriumPhase> parseEquilibriumPhases(
+    boost::optional<BaseLib::ConfigTree> const& config)
+{
+    if (!config)
+        return {};
+
+    std::vector<EquilibriumPhase> nodal_equilibrium_phases;
+    for (
+        auto const& equilibrium_phase_config :
+        //! \ogs_file_param{prj__chemical_system__equilibrium_phases__equilibrium_phase}
+        config->getConfigSubtreeList("equilibrium_phase"))
+    {
+        //! \ogs_file_param{prj__chemical_system__equilibrium_phases__equilibrium_phase__name}
+        auto const name =
+            equilibrium_phase_config.getConfigParameter<std::string>("name");
+
+        //! \ogs_file_param{prj__chemical_system__equilibrium_phases__equilibrium_phase__initial_amount}
+        double const initial_amount =
+            equilibrium_phase_config.getConfigParameter<double>(
+                "initial_amount");
+
+        //! \ogs_file_param{prj__chemical_system__equilibrium_phases__equilibrium_phase__saturation_index}
+        double const saturation_index =
+            equilibrium_phase_config.getConfigParameter<double>(
+                "saturation_index");
+
+        nodal_equilibrium_phases.emplace_back(
+            name, initial_amount, saturation_index);
+    }
+
+    return nodal_equilibrium_phases;
+}
+
+std::ofstream& operator<<(
+    std::ofstream& outFile,
+    std::vector<EquilibriumPhase> const& equilibrium_phases)
+{
+    for (auto const& equilibrium_phase : equilibrium_phases)
+    {
+        outFile << equilibrium_phase._name.c_str();
```

Comment:
no need to convert a string to char array. There is an operator<<(std::string) overload...

Reply:
Learned. Fixed.

Reply:
One more place in the ReactionRate.cpp:57

Reply:
Good catch. Fixed.